# Generated by Django 5.2.2 on 2025-08-15 15:16

import django.db.models.deletion
from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='UserProfile',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('phone_number', models.CharField(blank=True, max_length=13, null=True, unique=True)),
                ('telegram_id', models.CharField(blank=True, max_length=10, null=True)),
                ('address', models.CharField(blank=True, max_length=60)),
                ('company', models.CharField(blank=True, max_length=60)),
                ('avatar', models.ImageField(blank=True, null=True, upload_to='avatars/')),
                ('bio', models.TextField(blank=True)),
                ('language', models.CharField(default='uk', max_length=5)),
                ('timezone', models.CharField(default='Europe/Kiev', max_length=50)),
                ('is_2fa_enabled', models.BooleanField(default=False)),
                ('totp_secret', models.CharField(blank=True, max_length=16, null=True)),
                ('is_subscribed_to_newsletter', models.BooleanField(default=True)),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('last_login', models.DateTimeField(auto_now=True)),
                ('failed_otp_attempts', models.IntegerField(default=0)),
                ('otp_lockout_until', models.DateTimeField(blank=True, null=True)),
                ('user', models.OneToOneField(on_delete=django.db.models.deletion.CASCADE, related_name='profile', to=settings.AUTH_USER_MODEL)),
            ],
        ),
    ]
